### PR TITLE
Ham-four integration testing with the message pump

### PR DESCRIPTION
**Pull request by ScottDugas**
Unfortunately these cannot be split up easily because I couldn't know if the message pump was good without changing the integration test, and couldn't write the integration test without the message pump. oh well.

### CODE REVIEW

File: src/test/java/com/ripariandata/timberwolf/hbase/HBaseMailWriterTest.java [hunk 1]
```diff
@@ -143,7 +143,7 @@ public void testInterfaces()
         MockHTable mockHTable = MockHTable.create("defaultTableName");
         String tableName = Bytes.toString(mockHTable.getTableName());
 
-        IHBaseTable table = new HBaseTable(mockHTable);
+        HBaseTable table = new HBaseTable(mockHTable);
```
Comment (skermes) [inline src/test/java/com/ripariandata/timberwolf/hbase/HBaseMailWriterTest.java:146]: Why does this need the concrete class?  Shouldn't everything except HBaseManager be using the interface?

File: src/test/java/com/ripariandata/timberwolf/integrated/HTableResource.java [hunk 2]
```diff
@@ -57,8 +67,14 @@ public Statement apply(final Statement inner, final Description description)
             @Override
             public void evaluate() throws Throwable
             {
-                hbaseManager = new HBaseManager(getProperty(ZOO_KEEPER_QUORUM_PROPERTY_NAME),
-                                                getProperty(ZOO_KEEPER_CLIENT_PORT_PROPERTY_NAME));
+                synchronized (MANAGER_LOCK)
```
Comment (skermes) [inline src/test/java/com/ripariandata/timberwolf/integrated/HTableResource.java:70]: What is this protecting against?  Two people on the same box running the tests at the same time?

File: src/test/java/com/ripariandata/timberwolf/integrated/HTableResource.java [hunk 3]
```diff
@@ -116,7 +134,7 @@ public String getFamily()
     }
 
     /**
-     * The table as created by our production code.
+     * The table as created by ice.submit(AbstractExour production code.
```
Comment (skermes) [inline src/test/java/com/ripariandata/timberwolf/integrated/HTableResource.java]: I feel like someone's ctrl+v may have slipped.

File: src/test/resources/log4j.xml [hunk 4]
```diff
@@ -31,19 +25,34 @@
...
+    <logger name="com.ripariandata.timberwolf.exchange.UserIterator">
+      <level value="error" />
+    </logger>
+
     <!-- ExchangeService and FindFolderIterator are particularly
          chatty during the integration test -->
     <logger name="com.ripariandata.timberwolf.exchange.ExchangeService">
       <!-- set this to trace to see all exchange http traffic -->
-      <level value="fatal" />
+      <level value="trace" />
```
Comment (skermes) [inline src/test/resources/log4j.xml]: Whoops, that's my bad.  I comitted it so I could get a look at the logs on Jenkins.  We probably want to turn it back to fatal.

Or maybe not?  I end up setting this to trace almost every time I'm debugging ExchangeService.

File: src/test/java/com/ripariandata/timberwolf/hbase/HBaseMailWriterTest.java (hunk 1 above)
Comment (ScottDugas) [inline src/test/java/com/ripariandata/timberwolf/hbase/HBaseMailWriterTest.java:146]: This table is passed to HBaseManager.addTable which requires an HBaseTable

File: src/test/java/com/ripariandata/timberwolf/integrated/HTableResource.java (hunk 2 above)
Comment (ScottDugas) [inline src/test/java/com/ripariandata/timberwolf/integrated/HTableResource.java:70]: We don't right now, but you can run jUnit tests in parallel, in which case these rules could be executed at the same time, so Im' just going to leave it in
http://stackoverflow.com/questions/423627/running-junit-tests-in-parallel

**Pull request closed without merging**